src/main/java/jared/task/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package jared.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+    protected LocalDate date;
+    protected LocalTime time;
+
+    /**
+     * Constructor for deadline task if date is provided.
+     * @param description of deadline task.
+     * @param date of deadline task.
+     */
+    public Deadline(String description, LocalDate date) {
+        super(description);
+        this.date = date;
+    }
+
+    /**
+     * Constructor for deadline task if time and date is provided.
+     * @param description of deadline task.
+     * @param date of deadline task.
+     * @param time of deadline task.
+     */
+    public Deadline(String description, LocalDate date, LocalTime time) {
+        super(description);
+        this.date = date;
+        this.time = time;
+    }
+
+    /**
+     * Converts to string for printing list.
+     * @return String of deadline details.
+     */
+    @Override
+    public String toString() {
+        String res = "[D]" + super.toString() + " (by: "
+                + date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
```

Comment:
the format of "d MMM" doesn't seem very intuitive to me, can consider changing to "dd MM"